### ISSUE TITLE
Swagger-play simple usage advice required - limitations or bug? 

### ISSUE THREAD

**Issue by sundriver** [swagger-api/swagger-play#29]
Hi there,
 Thanks for developing swagger-play, it is quite useful. We are using 1.3.13 version with play 2.4.4.

 I have issue trying to generate correct swagger code for the following simple model. It is likely I am missing something but since I couldn't figure out what, I thought of posting it here for advice.

case class Application2(
  @(ApiModelProperty @field)(required=true) audit:Audit2
)

case class Audit2(
  @(ApiModelProperty @field)(required=true) createdAt:String
)

@Api(value="/rest/applications", description="Applications")
class ApplicationController extends Controller {
   @ApiOperation(
    nickname = "getApplication",
    produces = "application/json",
    response = classOf[Application2],
    value = "Gets an application"
  )
  def application = Action{r=>Ok("test")}
}

For the route /rest/applications above, this is producing:

Application2 {
audit (object, optional)
}

in other words:
{
  "audit": {}
}

Issues:
- it shows the attribute as object, I was expecting to see swagger-play to also show the structure of Audit2 case class. 
- It is uses optional while I clearly set the required attribute to true

Could you please advise how these issues can be fixed. Are these bugs or limitation of swagger-play library?

Thanks,
Ali

**Comment by fehguy** [comment]
hi @sundriver, just now (as in, within the last 10 minutes), we updated the play2 library for play 2.4 + swagger-spec 2.0.  Can you please take a look at the sample here:

[scala-play2.4](https://github.com/swagger-api/swagger-samples/tree/master/scala/scala-play2.4)

which leverages this new library implementation:

[swagger-play2](https://github.com/swagger-api/swagger-play/tree/master/play-2.4/swagger-play2)

**Comment by sundriver** [comment]
This is perfect, thanks guys, all working now. I am closing the issue
